utils/src/dense_polynomial.rs

```diff
@@ -28,9 +29,9 @@ pub trait ExtendedDensePolynomial<F: Field> {
 impl<F: Field> ExtendedDensePolynomial<F> for DensePolynomial<F> {
     fn scale(&self, elm: F) -> Self {
         let mut result = self.clone();
-        for coeff in &mut result.coeffs {
+        result.coeffs.par_iter_mut().for_each(|coeff| {
```

Comment:
I think `for coeff in result.coeffs.par_iter_mut() {` should work no?

Reply:
Doesn't work actually, only works for `Iterator`s, not the parallel iterator trait from rayon.